pkg/scheduler/apis/config/v1beta3/default_plugins.go

```diff
@@ -122,17 +67,23 @@ func getDefaultPlugins() *v1beta3.Plugins {
 
 func applyFeatureGates(config *v1beta3.Plugins) {
 	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeCapacityPriority) {
-		config.Score.Enabled = append(config.Score.Enabled, v1beta3.Plugin{Name: names.VolumeBinding, Weight: pointer.Int32Ptr(1)})
+		// Remove VolumeBinding from Score.Disabled if the feature gate is enabled
+		// Since VolumeBinding is the only disabled plugin right now in the default registry, we can just set Disabled to nil
+		config.Score.Disabled = nil
 	}
 
 	if !utilfeature.DefaultFeatureGate.Enabled(features.DefaultPodTopologySpread) {
 		// When feature is enabled, the default spreading is done by
 		// PodTopologySpread plugin, which is enabled by default.
 		klog.InfoS("Registering SelectorSpread plugin")
-		s := v1beta3.Plugin{Name: names.SelectorSpread}
-		config.PreScore.Enabled = append(config.PreScore.Enabled, s)
-		s.Weight = pointer.Int32Ptr(1)
-		config.Score.Enabled = append(config.Score.Enabled, s)
+		s := v1beta3.Plugin{Name: names.SelectorSpread, Weight: pointer.Int32Ptr(1)}
+		for index, plugin := range config.MultiPoint.Enabled {
+			if plugin.Name == names.PodTopologySpread {
+				config.MultiPoint.Enabled[index] = s
+				config.MultiPoint.Disabled = append(config.MultiPoint.Disabled, v1beta3.Plugin{Name: names.PodTopologySpread})
```

Comment:
I don't understand this, why are we disabling the whole PodTopologySpread plugin, that doesn't sound right. We only need to enable `SelectorSpread`

Reply:
This was something I noticed while reordering the Filter plugins above, and maybe I misunderstood it. But if the `DefaultPodTopologySpread` feature gate is disabled, shouldn't that disable PodTopologySpread as a default plugin as well? Looking at it again, we weren't doing that before so maybe that's not the intent. I can revert this part if so

Reply:
We need to revert it, when this feature is enabled, we should simply add the plugin. We don't want to completely disable `PodTopologySpread`, we only want to skip default spreading via PodTopologySpread which is done here https://github.com/kubernetes/kubernetes/blob/9b45983d3c8c246948682e678caa721b6d0bed0c/pkg/scheduler/apis/config/v1beta3/defaults.go#L230

Reply:
Okay, will do then